### PR TITLE
added diag

### PR DESCRIPTION
**Pull request by atteson**

### CODE REVIEW

File: src/woodbury.jl [hunk 1]
```diff
@@ -76,6 +76,17 @@ function inv(W::AbstractWoodbury)
     return Woodbury(A′, U′, C′, V′)
 end
 
+# multiplying everything out would be O(m^2 k + m k^2) operations
+# best I could think of was O(m^2 k) operations
+function diag(W::AbstractWoodbury)
+    d = diag(W.A)
+    (k,m) = size(W.U)
+    for i = 1:k
+        d[i] += W.U[i,:]' * W.C * W.V[:,i]
```
Comment (timholy) [inline src/woodbury.jl]: Would
```julia
d = diag(W.A) + vec(sum(W.U .* (W.C * W.V)', dims=2))
```
be faster? Matrix-matrix multiplication is optimized more than the collection of matrix-vector products, and your version also copies each column of `U` and `V`.

**Review by timholy — COMMENTED** [review]
Sorry I failed to notice this. This looks like good functionality, but see below re implementation.

File: src/woodbury.jl (hunk 1 above)
Comment (atteson) [reply src/woodbury.jl]: I tested this on some large matrices.  I've seen it go twice as fast empirically for large matrices though my math suggests that it should be the same.  It's also more concise.

This is my first pull request.  I pushed the change into my branch.  As far as I can tell, I don't need to do anything else but would be happy to if needed.

**Comment by timholy** [comment]
Thanks @atteson!

**Pull request merged**